src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import warnings
+from typing import List, Optional, Sequence, TypeVar
+
+from sqlalchemy import RowMapping, text
+from sqlalchemy.exc import ProgrammingError
+
+from ..engine import AlloyDBEngine
+
+COLLECTIONS_TABLE = "langchain_pg_collection"
+EMBEDDINGS_TABLE = "langchain_pg_embedding"
+
+T = TypeVar("T")
+
+
+async def _aget_collection_uuid(
+    engine: AlloyDBEngine,
+    collection_name: str,
+) -> str:
+    """
+    Get the collection uuid for a collection present in PGVector tables.
+
+    Args:
+        collection_name (str): The name of the collection to get the uuid for.
+    Returns:
+        The uuid corresponding to the collection.
+    """
+    query = (
+        f"SELECT name, uuid FROM {COLLECTIONS_TABLE} WHERE name = '{collection_name}'"
+    )
+    async with engine._pool.connect() as conn:
+        result = await conn.execute(text(query))
+        result_map = result.mappings()
+        result_fetch = result_map.fetchone()
+    if not result_fetch:
+        raise ValueError(f"Collection, {collection_name} not found.")
+    return result_fetch.uuid
+
+
+async def _aextract_pgvector_collection(
+    engine: AlloyDBEngine,
+    collection_name: str,
+) -> Sequence[RowMapping]:
+    """
+    Extract all data belonging to a PGVector collection.
+
+    Args:
+        collection_name (str): The name of the collection to get the data for.
+
+    Returns:
+        The data present in the collection.
+    """
+    uuid = await _aget_collection_uuid(engine, collection_name)
+    try:
+        query = f"SELECT * FROM {EMBEDDINGS_TABLE} WHERE collection_id = '{uuid}'"
```

Comment:
This doesn't scale for large datasets because we are pulling everythign into memory. We could use the loader with lazy load to load in batches to insert.

Reply:
Implemented data batching. Data loader loads data in document form so this is more straightforward and efficient.